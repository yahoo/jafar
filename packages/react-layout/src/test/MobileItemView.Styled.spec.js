/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import renderer from 'react-test-renderer';
import Style from '../components/MobileItemView/Styled';
import 'jest-styled-components';

describe('Styled', () => {
  [
    { size: 1 },
    { size: 2 },
    { size: 3 },
    { size: 4 },
  ]
    .forEach(({ size }) => {
      const Styled = Style(size);
      describe(`size: ${size}`, () => {
        it('Wrapper', () => {
          const tree = renderer.create(<Styled.Wrapper />).toJSON();
          expect(tree).toMatchSnapshot();
        });
        it('Sections', () => {
          const tree = renderer.create(<Styled.Sections />).toJSON();
          expect(tree).toMatchSnapshot();
        });
        it('Header', () => {
          const tree = renderer.create(<Styled.Header />).toJSON();
          expect(tree).toMatchSnapshot();
        });
        it('Title', () => {
          const tree = renderer.create(<Styled.Title />).toJSON();
          expect(tree).toMatchSnapshot();
        });
      });
    });
});
