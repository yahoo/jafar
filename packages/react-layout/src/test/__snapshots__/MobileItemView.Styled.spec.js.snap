// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled size: 1 Header 1`] = `
.c0 {
  background-color: #f0f0f0;
  font-size: 16px;
  padding: 10px 10px 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 1 Sections 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 1 Title 1`] = `
.c0 {
  font-size: 18px;
  font-weight: bold;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 45px;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 1 Wrapper 1`] = `
.c0 {
  color: #3d3d3d;
  background-color: #fff;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2 Header 1`] = `
.c0 {
  background-color: #f0f0f0;
  font-size: 16px;
  padding: 10px 10px 10px 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2 Sections 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2 Title 1`] = `
.c0 {
  font-size: 18px;
  font-weight: bold;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 45px;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2 Wrapper 1`] = `
.c0 {
  color: #3d3d3d;
  background-color: #fff;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3 Header 1`] = `
.c0 {
  background-color: #f0f0f0;
  font-size: 16px;
  padding: 10px 10px 10px 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3 Sections 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3 Title 1`] = `
.c0 {
  font-size: 18px;
  font-weight: bold;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 45px;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3 Wrapper 1`] = `
.c0 {
  color: #3d3d3d;
  background-color: #fff;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4 Header 1`] = `
.c0 {
  background-color: #f0f0f0;
  font-size: 16px;
  padding: 10px 10px 10px 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4 Sections 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4 Title 1`] = `
.c0 {
  font-size: 18px;
  font-weight: bold;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 45px;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4 Wrapper 1`] = `
.c0 {
  color: #3d3d3d;
  background-color: #fff;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;
